plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java

```diff
@@ -891,6 +931,31 @@ public String toString() {
                     diskBuilder.append("<read_iops_sec>" + _iopsReadRate + "</read_iops_sec>\n");
                 if ((_iopsWriteRate != null) && (_iopsWriteRate > 0))
                     diskBuilder.append("<write_iops_sec>" + _iopsWriteRate + "</write_iops_sec>\n");
+                if (s_qemuVersion >= 2004000) {
+                    if (_bytesReadRateMax != null && _bytesReadRateMax > 0 ) {
+                        diskBuilder.append("<read_bytes_sec_max>" + _bytesReadRateMax + "</read_bytes_sec_max>\n");
+                    }
+                    if (_bytesWriteRateMax != null && _bytesWriteRateMax > 0) {
+                        diskBuilder.append("<write_bytes_sec_max>" + _bytesWriteRateMax + "</write_bytes_sec_max>\n");
+                    }
+                    if (_iopsReadRateMax != null && _iopsReadRateMax > 0)
+                        diskBuilder.append("<read_iops_sec_max>" + _iopsReadRateMax + "</read_iops_sec_max>\n");
+                    if (_iopsWriteRateMax != null && _iopsWriteRateMax > 0)
+                        diskBuilder.append("<write_iops_sec_max>" + _iopsWriteRateMax + "</write_iops_sec_max>\n");
+                }
+                if (s_qemuVersion >= 2006000) {
```

Comment:
I like, however/and; It might be wise not to check on the number here but asses whether the feature is supported as more then once features have been added and disappeared for security reasons from smantically newer/older versions. I suggest a versionsSupports<Feature>(version) set of functions to be called in this kind of conditionals. i.e.
versionSupportsBursts(version) and do the switch/regexp/simple conditional in it.
Most of the times a simple > minimal version will work but in libvirt, or in for instance bouncy castle that is not always true.

Reply:
@DaanHoogland suggestion, looks a great way of managing differnet versions and their supported actions. I am +1 on creating such method.

Another approach that also helps (but less elaborated than the method) is extracting the version value to a variable with a suggestive name and a Javadoc documenting the version :wink: